File: hw/Rodriguez/u2/hw17/ComplexOperation/ec_edu_espe_complexoperation_controller/person_controller.py
# person_controller.py
from datetime import datetime, date

class PersonController:
    @staticmethod
    def compute_age_in_days(birth_date):
        today = date.today()
        period = today - birth_date
        return period.days

    @staticmethod
    def compute_age_in_years(birth_date):
        today = date.today()
        return today.year - birth_date.year - ((today.month, today.day) < (birth_date.month, birth_date.day))

File: hw/segarra/u2/HW17TestCaseSolution/ComplexOperationsJavaScript/ec.edu.espe.complexoperationsjavascript.model/PersonController.js
const { differenceInDays, differenceInYears } = require('date-fns');

function computeAgeInDays(birthDate) {
    const today = new Date(2024, 7, 11, 0, 1);
    const [day, month, year] = birthDate.split('-').map(Number);
    const birth = new Date(year, month - 1, day, 0, 1);
    return differenceInDays(today, birth);
}

function computeAgeInYears(birthDate) {
    const today = new Date(2024, 7, 11, 0, 1);
    const [day, month, year] = birthDate.split('-').map(Number);
    const birth = new Date(year, month - 1, day, 0, 1);
    return differenceInYears(today, birth);
}

module.exports = computeAgeInDays
module.exports = computeAgeInYears